src/filesystem/FileSystem.js

```diff
@@ -851,11 +852,19 @@ define(function (require, exports, module) {
      * @param {function(string): boolean} filter - Returns true if a particular item should
      *      be watched, given its name (not full path). Items that are ignored are also
      *      filtered from Directory.getContents() results within this subtree.
+     * @param {string} globFilter - glob compatible string definition for filtering out events
+     *      on the node side.
      * @param {function(?string)=} callback - A function that is called when the watch has
      *      completed. If the watch fails, the function will have a non-null FileSystemError
      *      string parametr.
      */
-    FileSystem.prototype.watch = function (entry, filter, callback) {
+    FileSystem.prototype.watch = function (entry, filter, globFilter, callback) {
+        // make globFilter an optional argument
+        if (typeof globFilter === "function" && typeof callback !== "function") {
+            callback = globFilter;
+            globFilter = null;
+        }
```

Comment:
Is this logic for backward compatibility? If so add a comment.
(I always find this kind of things hard to follow...)
Is the second condition necessary?